src/NUnitFramework/nunitlite/CommandLineOptions.cs

```diff
@@ -64,18 +66,102 @@ internal CommandLineOptions(IDefaultOptionsProvider defaultOptionsProvider, para
             
             ConfigureOptions();            
             if (args != null)
-                Parse(args);
+                Parse(PreParse(args));
         }
 
         public CommandLineOptions(params string[] args)
         {
             ConfigureOptions();
             if (args != null)
-                Parse(args);
+                Parse(PreParse(args));
         }
-        
+
+#if !PORTABLE
+        private int _nesting = 0;
+#endif
+
+        internal IEnumerable<string> PreParse(IEnumerable<string> args)
+        {
+#if PORTABLE
+            return args;
+#else
+            if (++_nesting > 3)
+            {
+                ErrorMessages.Add("@ nesting exceeds maximum depth of 3");
+                --_nesting;
+                return args;
+            }
+
+            var listArgs = new List<string>();
+            var delim = ' ';
+
+            foreach (var arg in args)
+            {
+                if (arg.Length == 0 || arg[0] != '@')
+                {
+                    listArgs.Add(arg);
+                    continue;
+                }
+
+                if (arg.Length == 1)
+                {
+                    ErrorMessages.Add("The file name should not be empty");
+                    continue;
+                }
+
+                var offset = 1;
+                if (arg.Length > 4 && arg[1] == '[' && arg[3] == ']')
+                {
+                    delim = arg[2];
+                    offset = 4;
+                }
+
+                var filename = arg.Substring(offset);
+
+                if (!File.Exists(filename))
+                {
+                    ErrorMessages.Add("The file \"" + filename + "\" was not found");
+                    continue;
+                }
+
+                string contents;
+                try
+                    {
+#if NETCF
+                    using (var sr = new StreamReader (filename))
+                        {
+                        contents = sr.ReadToEnd ();
+                        }
+#else
+                    contents = File.ReadAllText (filename);
+#endif
+                    }
+                catch (IOException ex)
+                    {
+                    ErrorMessages.Add("Error reading \"" + filename + "\": " + ex.Message);
+                    continue;
+                    }
+
+                var newArgs = GetArgs(contents.Replace("\r", "").Replace('\n', delim));
+                listArgs.AddRange(PreParse(newArgs));
+            }
+
+            --_nesting;
+            return listArgs;
+#endif
+        }
+
+#if !PORTABLE
+        internal static IEnumerable<string> GetArgs(string commandLine)
+        {
+            const string re = @"\G(""((""""|[^""])+)""|(\S+)) *";
+            var ms = Regex.Matches(commandLine, re);
+            return ms.Cast<Match>().Select(m => Regex.Replace(m.Groups[2].Success ? m.Groups[2].Value : m.Groups[4].Value, @"""""", @""""));
+        }
```

Comment:
Your code seems cleaner, but it bothers me that we have two entirely different implementations of this method, here and in https://github.com/nunit/nunit-console/pull/98/files. I do prefer using the named groups in the other rather than positional.

BTW, I hate code reviewing RegEx 😝